dynamic_programming/longest_common_subsequence.cpp

```diff
@@ -0,0 +1,287 @@
+/**
+ * @file
+ * @author [Santosh Jonnakuti](https://github.com/Santoshjonnakuti)
+ * @brief [Longest Common Subsequence Algorithm](https://en.wikipedia.org/wiki/Longest_common_subsequence_problem)
+ * implementation in C to find the LCS between 2 strings.
+ * @details LCS algorithm uses dynamic programming approach to find all possible LCS between 2 strings. 
+ * Dynamic Programming follow a method so that even sub-problems will also have optimal solution.
+ *
+ * LCS() will find the length of the LCS of the following two strings 'GAC' and 'AGCAT'.
+ * LCS function will return the following 2D-Matrix
+ *
+ *	    0 0 0 0 0 0
+ *      0 0 1 1 1 1
+ *      0 1 1 1 2 2
+ *      0 1 1 2 2 2
+ *
+ * So Length of LCS will be 2.
+ *
+ * allPossibleLSC() will print all possible LCS between those strings.
+ *
+ * AC
+ * GC
+ * GA
+ * GA
+ * 
+ * So There are 3 Longest common subsequences in 'GAC' and 'AGCAT'
+ */
+
+
+#include<stdio.h>     /// for I/O operations
+#include<stdlib.h>   ///  for Dynamic memory allocation
+#include<string.h>  ///   for puts(), gets() and strlen()
+
+#define MAX 100      // maximum size of string1 and string2
+
+int lengthOfLCS;      // to store length of LCS 
+char **LCSArray;      // 2D- Character array to store all possible LCS
+int ind=0;
```

Comment:
Avoid using global variables as they may cause overflows and other issues. Pass them as local function parameters or as an structure members.